### PR TITLE
New item: the bluespace razor

### PR DESCRIPTION
**Pull request by Tharcoonvagh**
## About The Pull Request
Adds a new item, the bluespace razor
basically a razor but faster, has a different suicide text and can shave people from a distance, since "bleewspaec"
(i might have hecked something up accidentally)

## Why It's Good For The Game
minor crew content!

## Changelog
:cl:tharcoonvagh
add: Bluespace razor
/:cl:

_this fucking pr....._

### CODE REVIEW

**Comment by ishitbyabullet** [comment]
bald buff

**Comment by wesoda25** [comment]
bruh just make it a path of normal razors

**Comment by wesoda25** [comment]
Also it should allow for specific styling without need for a mirror

**Comment by itseasytosee** [comment]
Why must it be blue space related? "bluespace energy" means nothing could you have not called it laser razor or anything more creative than tacking on bluespace

**Comment by Shadowflame909** [comment]
its my suggestion nerd

if you have beef take it up with me in the NTR hut
Its where you'll find me

Also was ranged balding too over powered

**Comment by ATH1909** [comment]
IMO, a bluespace razor should act like a normal razor, except you don't need to remove all articles of clothing from someone's head and/or mouth to use it on them (still restrict it to only being usable from 1 space away or less, though). _That_ would be a somewhat useful pranking item.

**Comment by Tharcoonvagh** [comment]
I see

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -1067,6 +1067,17 @@
 	hidden = TRUE
 	experimental = TRUE
 
+/datum/techweb_node/bluerazor
```
[the remark this answers is not in the document]
Comment (Tharcoonvagh) [reply code/modules/research/techweb/all_nodes.dm]: Aight

**Comment by Tlaltecuhtli** [comment]
make it work on people withing 8 tiles

**Comment by Tharcoonvagh** [comment]
How exactly should I do that?

**Comment by ArcaneMusic** [comment]
Well, for starters, you're going to want to make this whole thing work off the existing razor procs as a subtype first.  Most of what you have this razor doing is identical to the existing areas, but the areas with differences can be checked by variables. EG: Give the base razor a variable like "bluespace_cutting" and for the default razor, it'll default to FALSE, for this one it'll default to true. From there, just have the areas where the razor checks for adjacency also check for bluespace, and if true it'll ignore the adjacency check. For more specific help you can hop onto the coding channel of the discord or the coderbus discord.

**Comment by Tharcoonvagh** [comment]
Its a subtype now, rejoice

**Comment by ATH1909** [comment]
> make it work on people within 8 tiles

No, don't do this. Razors get blocked by pretty much any head item except for stuff for your ears (and maybe your eyes?), so you need to get close enough to someone to strip off their headgear anyway. Also, making it work on people within 8 tiles _and_ making it work through articles of clothing would make it kind of hard to counter if you don't know what a bluespace razor is/haven't seen one before and aren't constantly moving (think of the old anti-hivemind gameplay).

**Comment by ATH1909** [comment]
> Its a subtype now, rejoice

Uh, no it isn't?

File: code/game/objects/items/cosmetics.dm [hunk 2]
```diff
@@ -107,6 +107,8 @@
 	flags_1 = CONDUCT_1
 	w_class = WEIGHT_CLASS_TINY
 
+/var/proximity = TRUE
```
Comment (ATH1909) [inline code/game/objects/items/cosmetics.dm]: bruh

File: code/game/objects/items/cosmetics.dm [hunk 3]
```diff
@@ -226,3 +228,142 @@
...
+		var/location = user.zone_selected
+		if((location in list(BODY_ZONE_PRECISE_EYES, BODY_ZONE_PRECISE_MOUTH, BODY_ZONE_HEAD)) && !H.get_bodypart(BODY_ZONE_HEAD))
+			to_chat(user, "<span class='warning'>[H] doesn't have a head!</span>")
+			return
+		if(location == BODY_ZONE_PRECISE_MOUTH)
+			if(user.a_intent == INTENT_HELP)
+				if(H.gender == MALE)
+
+				if(!proximity && prox_check)
+        return
```
Comment (ATH1909) [inline code/game/objects/items/cosmetics.dm]: bruh

File: code/game/objects/items/cosmetics.dm [hunk 4]
```diff
@@ -226,3 +228,142 @@
...
+						user.visible_message("<span class='warning'>[user] shaves off [H]'s facial hair with [src].</span>", \
+											 "<span class='notice'>You shave [H]'s facial hair clean off, though his skin is a bit burnt from the bluespace sparks.</span>")
+						shave(H, location)
+
+		else if(location == BODY_ZONE_HEAD)
+
+		if(!proximity && prox_check)
+        return
+
+			if(user.a_intent == INTENT_HELP)
```
Comment (ATH1909) [inline code/game/objects/items/cosmetics.dm]: this probably isn't supposed to be indented

File: code/game/objects/items/cosmetics.dm [hunk 5]
```diff
@@ -226,3 +228,142 @@
...
+		var/mob/living/carbon/human/H = M
+		var/location = user.zone_selected
+		if((location in list(BODY_ZONE_PRECISE_EYES, BODY_ZONE_PRECISE_MOUTH, BODY_ZONE_HEAD)) && !H.get_bodypart(BODY_ZONE_HEAD))
+			to_chat(user, "<span class='warning'>[H] doesn't have a head!</span>")
+			return
+		if(location == BODY_ZONE_PRECISE_MOUTH)
+			if(user.a_intent == INTENT_HELP)
+				if(H.gender == MALE)
+
+				if(!proximity && prox_check)
```
Comment (ATH1909) [inline code/game/objects/items/cosmetics.dm]: bruh

File: code/game/objects/items/cosmetics.dm (hunk 3 above)
Comment (ATH1909) [reply code/game/objects/items/cosmetics.dm]: (properly indent this return)

File: code/game/objects/items/cosmetics.dm (hunk 2 above)
Comment (Tharcoonvagh) [reply code/game/objects/items/cosmetics.dm]: what do you suggest then? im open for anything

File: code/game/objects/items/cosmetics.dm (hunk 3 above)
Comment (Tharcoonvagh) [reply code/game/objects/items/cosmetics.dm]: same here

**Comment by ATH1909** [comment]
To cut a long story (and a lot of bruhs) short:

- fix yer indentation
- fix yer prox_check variable (or just cut it out entirely and use one of the procs that probably already exist in the game for stuff like that instead)
- rename yer "proximity" variable to something like "requires_proximity" and put it with the other variables (also, I don't think that there's supposed to be a "/" in front of "var/")

**Comment by Tharcoonvagh** [comment]
> * fix yer indentation (there are a LOT of indentation errors in this, and I didn't add "bruh"s to even close to all of them)

wha
I'll just use the BRPED procs then i guess

**Comment by MCterra10** [comment]
this is based

**Comment by Tharcoonvagh** [comment]
what
is
a
MERGE CONFLICT

**Comment by Tharcoonvagh** [comment]
How can I fix this? Someone tell me please!
Also,
is it finally working now?
aaaaaaaaaa

**Comment by Tharcoonvagh** [comment]
whats wrong with the map diff? did I modify something? dont think so

**Comment by ArcaneMusic** [comment]
Well, the 2 files that you have a merge conflict on, as listed at the bottom of the PR, have been updated in-between when you made your branch and now. This only matters here because these are files that you're trying to change as well. Map Diff and Appveyor are failing because they don't want you to effectively revert an existing change unintentionally, basically. Fixing the conflict in misc_designs is fairly simple, and can even be done just as a web edit (basically pasting in the new changes from the document, in it's simplest form). However, as you also have a conflict with the .dmi file, that means you should probably do it the proper way, which is updating your master, and merging the upstream/master with your branch, so you keep your changes from your commits, but at the same time maintain your sprites/code.

**Comment by Tharcoonvagh** [comment]
Cool thanks!

**Comment by Tharcoonvagh** [comment]
fuck what have I done

**Comment by Tharcoonvagh** [comment]
what the fuck

**Comment by Tharcoonvagh** [comment]
this is literally hell itself
I'VE DOWNLOADED THE RAW FILE, INSERTED MY ICON THEN COPIED IT TO THE AAAAAAAAA

**Comment by Tharcoonvagh** [comment]
okay my bluerazor icon somehow conflicts with something

**Comment by Tharcoonvagh** [comment]
why does it conflict with ANY merges

**Comment by Tharcoonvagh** [comment]
I added my icon to the dmi file and nothing else, someone help meeeeeee

**Comment by Tharcoonvagh** [comment]
Sorry @ninjanomnom but @ShizCalev doesnt seem interested in reviewing it, despite him being the "code owner"

**Comment by Couls** [comment]
oh my

**Comment by Arturlang** [comment]
Did you try updating your local by merging from upstream?

**Comment by Tharcoonvagh** [comment]
yes
twice

**Comment by Couls** [comment]
this doesn't compile
image

**Comment by Tharcoonvagh** [comment]
wah

File: code/game/objects/items/cosmetics.dm [hunk 6]
```diff
@@ -226,3 +226,138 @@
...
+
+/obj/item/razor/bluerazor/proc/blueshave(mob/living/carbon/human/H, location = BODY_ZONE_PRECISE_MOUTH)
+
+	if(location == BODY_ZONE_PRECISE_MOUTH)
+		H.facial_hairstyle = "Shaved"
+	else
+		H.hairstyle = "Skinhead"
+
+	H.update_hair()
+	playsound(loc, 'sound/items/welder2.ogg',  20, TRUE), (loc, 'sound/weapons/taser.ogg', 1, TRUE)//sparks
```
Comment (spookydonut) [inline code/game/objects/items/cosmetics.dm]: weird line, pick one of the sounds

File: code/game/objects/items/cosmetics.dm [hunk 7]
```diff
@@ -226,3 +226,138 @@
 			..()
 	else
 		..()
+
+/obj/item/razor/bluerazor
+	name = "bluespace razor"
+	desc = "A razor sparkling with bluespace energy."
+	icon = 'icons/obj/bluerazor.dmi'
+	icon_state = "bluerazor"
+	id = "bluerazor"
```
Comment (spookydonut) [inline code/game/objects/items/cosmetics.dm]: this is only for designs
```suggestion
```

File: code/modules/research/designs/misc_designs.dm [hunk 8]
```diff
@@ -77,6 +77,16 @@
 	category = list("Equipment")
 	departmental_flags = DEPARTMENTAL_FLAG_SCIENCE | DEPARTMENTAL_FLAG_ENGINEERING
 
+/datum/design/bluerazor
+	name = "Bluespace Razor"
+	desc = "A razor sparkling with bluespace energy."
+	id = "bluerazor"
+	build_type = PROTOLATHE
+	materials = list(/datum/material/iron = 600, /datum/material/bluespace = 300)
+	build_path = obj/item/razor/bluerazor
```
Comment (spookydonut) [inline code/modules/research/designs/misc_designs.dm]:
```suggestion
	build_path = /obj/item/razor/bluerazor
```

File: code/modules/research/designs/misc_designs.dm [hunk 9]
```diff
@@ -77,6 +77,16 @@
...
 	departmental_flags = DEPARTMENTAL_FLAG_SCIENCE | DEPARTMENTAL_FLAG_ENGINEERING
 
+/datum/design/bluerazor
+	name = "Bluespace Razor"
+	desc = "A razor sparkling with bluespace energy."
+	id = "bluerazor"
+	build_type = PROTOLATHE
+	materials = list(/datum/material/iron = 600, /datum/material/bluespace = 300)
+	build_path = obj/item/razor/bluerazor
+	cathegory = list("Equipment")
```
Comment (spookydonut) [inline code/modules/research/designs/misc_designs.dm]:
```suggestion
	category = list("Equipment")
```

**Comment by Tharcoonvagh** [comment]
Thanks @spookydonut!

**Comment by Tharcoonvagh** [comment]
Fuck

**Comment by Tharcoonvagh** [comment]
ill fix those, currently unable to code
just like the other errors
sorry, i'm really busy

**Comment by kriskog** [comment]
I'll put you out of your misery here. This really isnt something we need, and it absolutely should not be printable. From a cursory look at the code, you also seem to have a lot of copy pasted behaviour in new procs instead of overriding existing parent procs, so for an eventual new project look into avoiding that.

**Comment by Tharcoonvagh** [comment]
I accept my fate, thank you

**Pull request closed without merging**